sphinx/domains/c.py

```diff
@@ -2576,9 +2578,8 @@ def _parse_simple_type_specifier(self) -> Optional[str]:
         if self.match(_simple_type_specifiers_re):
             return self.matched_text
         for t in ('bool', 'complex', 'imaginary'):
-            if t in self.config.c_extra_keywords:
-                if self.skip_word(t):
-                    return t
+            if t in self.config.c_extra_keywords and self.skip_word(t):
+                return t
```

Comment:
Should not be changed, the original better conveys the semantics and is more readable.